src/global_distrib/mod_global_distrib_mapping_redis.erl

```diff
@@ -279,15 +285,17 @@ set_endpoints(Endpoints) ->
     refresh_set(EndpointsKey, BinEndpoints).
 
 -spec endpoint_to_binary(mod_global_distrib_utils:endpoint()) -> binary().
-endpoint_to_binary({IpAddr, Port}) ->
-    iolist_to_binary([inet:ntoa(IpAddr), "#", integer_to_binary(Port)]).
+endpoint_to_binary({IpAddr, Port}) when is_tuple(IpAddr) ->
+    iolist_to_binary([inet:ntoa(IpAddr), "#", integer_to_binary(Port)]);
+endpoint_to_binary({Domain, Port}) when is_list(Domain) ->
+    iolist_to_binary([list_to_binary(Domain), "#", integer_to_binary(Port)]).
```

Comment:
Since you already use `iolist_to_binary` here, you can drop `list_to_binary`.

Or, since domains might have unicode characters, replace `list_to_binary` with `unicode:characters_to_binary`